nvflare/private/fed/authenticator.py

```diff
@@ -0,0 +1,322 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import socket
+import time
+import traceback
+import uuid
+
+from nvflare.apis.fl_constant import ServerCommandKey
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.fl_exception import FLCommunicationError
+from nvflare.apis.shareable import Shareable
+from nvflare.apis.signal import Signal
+from nvflare.fuel.f3.cellnet.cell import Cell
+from nvflare.fuel.f3.cellnet.core_cell import make_reply as make_cellnet_reply
+from nvflare.fuel.f3.cellnet.defs import IdentityChallengeKey, MessageHeaderKey
+from nvflare.fuel.f3.cellnet.defs import ReturnCode
+from nvflare.fuel.f3.cellnet.defs import ReturnCode as F3ReturnCode
+from nvflare.fuel.f3.cellnet.fqcn import FQCN
+from nvflare.fuel.f3.message import Message
+from nvflare.fuel.f3.message import Message as CellMessage
+from nvflare.fuel.utils.log_utils import get_obj_logger
+from nvflare.private.defs import CellChannel, CellChannelTopic, CellMessageHeaderKeys, new_cell_message
+from nvflare.private.fed.utils.identity_utils import IdentityAsserter, IdentityVerifier, TokenVerifier, load_crt_bytes
+
+
+def _get_client_ip():
```

Comment:
fed.client.Communicator already has this function.

Reply:
Yes, moved.